cranelift/interpreter/src/step.rs

```diff
@@ -133,9 +142,17 @@ where
     // Interpret a Cranelift instruction.
     Ok(match inst.opcode() {
         Opcode::Jump | Opcode::Fallthrough => ControlFlow::ContinueAt(branch(), args()?),
-        Opcode::Brz => branch_when(!arg(0)?.into_bool()?)?,
-        Opcode::Brnz => branch_when(arg(0)?.into_bool()?)?,
-        Opcode::BrIcmp => branch_when(icmp(inst.cond_code().unwrap(), &arg(1)?, &arg(2)?)?)?,
+        Opcode::Brz => branch_when(match arg(0)?.ty() {
+            ty if ty.is_bool() => !arg(0)?.into_bool()?,
+            ty if ty.is_int() => arg(0)?.into_int()? == 0,
+            _ => return Err(StepError::ValueError(ValueError::InvalidValue(types::B1))),
+        })?,
+        Opcode::Brnz => branch_when(match arg(0)?.ty() {
+            ty if ty.is_bool() => arg(0)?.into_bool()?,
+            ty if ty.is_int() => arg(0)?.into_int()? != 0,
+            _ => return Err(StepError::ValueError(ValueError::InvalidValue(types::B1))),
+        })?,
```

Comment:
Hm, not totally sure about how to go about this; seems like there are a few options:
 - keep this as you implemented it, which is a bit more verbose
 - implement `into_bool` for integer types, but this may be a bit confusing because for other `into_*` functions we error when the type doesn't match
 - implement a `ValueConversionKind::ToBoolean` and just use `arg(0)?.convert(ValueConversionKind::ToBoolean)` here; I would probably lean this direction

Thoughts?

Reply:
> implement into_bool for integer types, but this may be a bit confusing because for other into_* functions we error when the type doesn't match

I actually started by doing this, but it didn't feel "right" so I changed it.

Yeah, I think the `ValueConversionKind` is a good way to go, I'll update the PR